shared/pkg/families/manager.go

```diff
@@ -77,17 +78,40 @@ func New(logger *log.Entry, config *Config) *Manager {
 
 type RunErrors map[types.FamilyType]error
 
-func (m *Manager) Run() (*results.Results, RunErrors) {
-	familyErrors := make(map[types.FamilyType]error)
+type familyResult struct {
+	result interfaces.IsResults
+	err    error
+}
 
+func (m *Manager) Run(ctx context.Context) (*results.Results, RunErrors) {
+	familyErrors := make(RunErrors)
 	familyResults := results.New()
 
 	for _, family := range m.families {
-		ret, err := family.Run(familyResults)
-		if err != nil {
-			familyErrors[family.GetType()] = fmt.Errorf("failed to run family %v: %w", family.GetType(), err)
-		} else {
-			familyResults.SetResults(ret)
+		result := make(chan familyResult)
+		go func() {
+			ret, err := family.Run(familyResults)
+			result <- familyResult{
+				ret,
+				err,
+			}
+		}()
```

Comment:
@chrisgacsal I'm not sure how the family run in parallel related to the current PR, but there is a problem here. There are families that depends on results from other families.
For example, the vulnerability family sometimes expect to get   `input_from_sbom`, How the parallel scan handle this?
Did you tried to run sbom & vulnerability together, the code is setting `familiesConfig.Vulnerabilities.InputFromSbom = true` in this case and I would expect to get a failure in the vulnerability family run.